### DISCUSSION TITLE
Thoughts on a ruby2js CLI

### DISCUSSION THREAD

**Discussion by jaredcwhite** [ruby2js/ruby2js#81] [Ideas]
I have a couple of questions/ideas which are somewhat related:

1. What do you think about including a `ruby2js` CLI in the gem? It could start with similar features/options as the script in the `demo` folder and maybe expand a bit. Which leads me to my second question:
2. For my Webpack loader, and even to build the loader itself (dogfooding FTW), I came up with a `rb2js.config.rb` format for setting up a loader with a bunch of configuration that gets used systematically. If we were to introduce a CLI in the ruby2js gem, I could see the `rb2js.config.rb` file format becoming formalized and maybe we even add a `ruby2js init` command to set up a new config file in the current direction. That way it feels similar to how many other compilation tools work (Webpack, Babel, PostCSS, etc.) What do you think?

**Comment by rubys** [comment]
I'm curious as to what features you feel belong in a CLI?  I will note that one can run `ruby demo/ruby2js.rb` from the command line today and it will either read Ruby from a file or from STDIN, and will write JavaScript to STDOUT.  I will say that from time to time I have felt the urge to create something with either the `-e` option of Ruby, or even better, a REPL like irb.

I have no current opinion on a `rb2js.config.rb` file (please read that as endorsing you to proceed as you feel best), but bringing this up makes me wonder if the Webpack loader be should be pulled into this repository?  It seems to me that the sprockets support should be retired in favor of this support.

**Reply by jaredcwhite** [reply]
Yeah so the idea would be that someone could install the gem and then immediately have the command line tool ready to roll, rather than having to pull down the repo with the demo folder.

As far as merging the loader into this repo, I'd be totally down with that! In fact, let me look into how I could give you push privileges for the NPM package.

**Comment by emilebosch** [comment]
👍  I thought it already came with a cli when I pulled it down and then I was like _aww._

**Reply by rubys** [reply]
Next release will have a CLI: https://github.com/ruby2js/ruby2js/commit/cc39ce46f6210532021f8c72416891b6edda69a1

**Comment by emilebosch** [comment]
Perfect!
Thanks so much!

**Comment by rubys** [comment]
Now released - v4.2.0
